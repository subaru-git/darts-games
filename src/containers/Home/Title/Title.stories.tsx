/* eslint-disable import/named */
import React from 'react';
import { ComponentMeta, ComponentStory } from '@storybook/react';
import Title from './Title';

export default {
  title: 'Components/Title',
  component: Title,
} as ComponentMeta<typeof Title>;

const Template: ComponentStory<typeof Title> = (args) => <Title {...args} />;

export const Default = Template.bind({});
Default.args = {};
