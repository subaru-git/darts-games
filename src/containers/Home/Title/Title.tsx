import React, { FC } from 'react';
import { Button, Center, Flex, Heading, Text } from '@chakra-ui/react';
import { css } from '@emotion/react';
import Router from 'next/router';
import useLocale from '@/hooks/locale';

const Title: FC = () => {
  const { t } = useLocale();
  return (
    <Flex
      direction='column'
      height={{ base: '220px', md: '400px' }}
      gap={{ base: 4, md: 8 }}
      justifyContent='center'
      css={style}
    >
      <Heading
        as={'h1'}
        size={{ base: 'lg', md: '2xl' }}
        textAlign='center'
        lineHeight='1.25em'
        whiteSpace='pre-wrap'
      >
        {t.home.title.title.join('\n')}
      </Heading>
      <Text fontSize={{ base: 'sm', md: 'lg' }} textAlign='center'>
        {t.home.title.description.join('\n')}
      </Text>
      <Center>
        <Button
          variant='outline'
          colorScheme='whatsapp'
          bg='#fefefe'
          width='fit-content'
          size={{ base: 'sm', md: 'md' }}
          onClick={() => Router.push('/games')}
        >
          {t.home.title.button}
        </Button>
      </Center>
    </Flex>
  );
};

const style = css({
  background: 'linear-gradient(#f5f5f5, #ededed, #f5f5f5)',
});
export default Title;
