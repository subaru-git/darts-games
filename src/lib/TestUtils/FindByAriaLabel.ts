export const findByAriaLabel = (elements: HTMLElement[], label: string) => {
  return (
    elements.find((button) => button.attributes.getNamedItem('aria-label')?.value === label) ??
    new HTMLElement()
  );
};
