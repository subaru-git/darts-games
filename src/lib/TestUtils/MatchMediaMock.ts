import mediaQuery from 'css-mediaquery'

export const createMatchMedia =
  (width: number) =>
  (query: string): MediaQueryList => ({
    matches: mediaQuery.match(query, { width }),
    media: query,
    onchange: null,
    addListener: jest.fn(), // deprecated
    removeListener: jest.fn(), // deprecated
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    dispatchEvent: jest.fn(),
  })
