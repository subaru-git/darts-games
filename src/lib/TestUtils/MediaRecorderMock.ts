export const MediaRecorderMock = {
  start: jest.fn(),
  ondataavailable: jest.fn(),
  onerror: jest.fn(),
  state: '',
  stop: jest.fn(),
};
