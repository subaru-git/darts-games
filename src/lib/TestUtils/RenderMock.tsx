import { ReactNode } from 'react';
import { ChakraProvider } from '@chakra-ui/react';
import { render as defaultRender } from '@testing-library/react';
import Providers from '@/contexts/Providers';

const render = (children: ReactNode | ReactNode[]) =>
  defaultRender(
    <ChakraProvider>
      <Providers>{children}</Providers>
    </ChakraProvider>,
  );
export { render };
