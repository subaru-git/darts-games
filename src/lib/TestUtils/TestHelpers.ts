import { waitFor as originalWaitFor, waitForOptions } from '@testing-library/react';

const defaultTimeout = process.env.WAIT_FOR_DEFAULT_TIMEOUT ? parseInt(process.env.WAIT_FOR_DEFAULT_TIMEOUT, 10) : 5000;

export async function waitFor(callback: () => void,  options?: waitForOptions): Promise<void> {
  return originalWaitFor(callback, { timeout: defaultTimeout, ...options });
}