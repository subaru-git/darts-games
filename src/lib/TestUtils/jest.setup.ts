import mocksdk from 'firebase-mock';
import failOnConsole from 'jest-fail-on-console';
import { createMatchMedia } from '@/lib/TestUtils/MatchMediaMock';
import { MediaRecorderMock } from '@/lib/TestUtils/MediaRecorderMock';

jest.mock('next/router', () => ({ useRouter: jest.fn().mockReturnValue({ locale: 'en' }) }));
jest.mock('react-camera-pro');
jest.mock('@/firebase/client', () => ({
  analytics: mocksdk.analytics,
  db: mocksdk.firestore,
  storage: mocksdk.storage,
  auth: mocksdk.auth,
  functions: mocksdk.functions,
}));
jest.mock('@firebase/auth');
beforeAll(() => {
  window.matchMedia = createMatchMedia(1100);
  window.MediaRecorder = (jest.fn() as any).mockImplementation(() => MediaRecorderMock);
  global.ResizeObserver = require('resize-observer-polyfill');
});

failOnConsole();
