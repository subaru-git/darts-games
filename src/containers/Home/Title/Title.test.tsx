import Title from './Title';
import { render } from '@/lib/TestUtils/RenderMock';

test('should rendering', () => {
  const { container } = render(<Title />);
  expect(container).toMatchSnapshot();
});
