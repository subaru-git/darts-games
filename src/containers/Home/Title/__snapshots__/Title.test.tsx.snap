// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should rendering 1`] = `
<div>
  <div
    class="flex h-52 flex-col justify-center gap-4 bg-gradient-to-b from-gray-100 via-gray-200 to-gray-100 md:h-80 md:gap-8"
  >
    <h1
      class="whitespace-pre-wrap text-center text-2xl font-bold leading-8 md:text-5xl"
    >
      It's dull throwing darts monotonously,
right?
    </h1>
    <span
      class="text-center text-sm md:text-lg"
    >
      It supports home darts, which tend to be monotonous.
    </span>
    <div
      class="flex justify-center"
    >
      <button
        class="text-emerald-600 border-emerald-600 bg-white hover:bg-emerald-50 disabled:opacity-50 w-fit rounded-lg border p-2 text-sm font-bold subpixel-antialiased disabled:cursor-not-allowed md:px-3 md:text-base undefined"
      >
        Let's see the practice games.
      </button>
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;
